cli/commands.go

```diff
@@ -437,11 +437,46 @@ func ImportImage(c *cli.Context) error {
 
 // AddNode adds a node to an existing cluster
 func AddNode(c *cli.Context) error {
+
+	/*
+	 * (0) Check flags
+	 */
+
 	clusterName := c.String("name")
-	nodeRole := c.String("role")
 	nodeCount := c.Int("count")
 
-	// check if cluster (i.e. the server) exists
+	/* (0.1)
+	 * --role
+	 * Role of the node that has to be created.
+	 * One of (server|master), (agent|worker)
+	 */
+	nodeRole := c.String("role")
+	if nodeRole == "worker" {
+		nodeRole = "agent"
+	}
+	if nodeRole == "master" {
+		nodeRole = "server"
+	}
+
+	// TODO: support this
+	if nodeRole == "server" {
+		return fmt.Errorf("ERROR: sorry, we don't support adding server nodes at the moment!")
+	}
+
+	/* (0.2)
+	 * --image, -i
+	 * The k3s image used for the k3d node containers
+	 */
+	image := c.String("image")
+	// if no registry was provided, use the default docker.io
+	if len(strings.Split(image, "/")) <= 2 {
```

Comment:
The shortest image I need to provide is something like rancher/k3s:v0.8.0 . -- so there should always be an "/" in the name. 

I am not sure if we need to any error checking here.  The wrong image name will be caught by the down stream logic.

Reply:
Fair points,  we can leave this logic as is for now.

Reply:
Fair points, we can leave this logic as is for now.